### PR TITLE
Add async_any_callback and async_callback macro with it's docs

### PR DESCRIPTION
**Pull request by shenjackyuanjie**
Do #395

### CODE REVIEW

File: socketio/src/asynchronous/mod.rs [hunk 1]
```diff
@@ -5,3 +5,48 @@ mod socket;
 #[cfg(feature = "async")]
 pub use client::builder::ClientBuilder;
 pub use client::client::Client;
+
+
+#[doc = r#"
+A macro to wrap a async callback function to be used in the client.
```
Comment (1c3t3a) [inline socketio/src/asynchronous/mod.rs:11]: "an async callback"

File: CHANGELOG.md [hunk 2]
```diff
@@ -34,10 +34,17 @@ The format is based on [Keep a Changelog], and this project adheres to
...
+
+_async callback macro_
+
+_2024.03.15_
+
+- Add macro `async_callback` and `async_any_callback` for async callbacks [#395](https://github.com/1c3t3a/rust-socketio/issue/395) [#399](https://github.com/1c3t3a/rust-socketio/pull/399)
 
 ## <a name="044">[0.4.4] - _Bump dependencies_ </a>
 
+
```
Comment (1c3t3a) [inline CHANGELOG.md:47]: Please remove this new line.

**Review by 1c3t3a — COMMENTED** [review]
LGTM % comments. Thanks a lot for this!

**Comment by 1c3t3a** [comment]
Seems like there is an error with formatting, please also fix that :)

For the failing test I will investigate what is going on.

**Comment by shenjackyuanjie** [comment]
I got it, it's from tokio's feature
I'll just add it and bump the version

**Comment by shenjackyuanjie** [comment]
good……
so now is the doc test issue

**Comment by 1c3t3a** [comment]
I fixed the tokio CI issue in main, feel free to rebase :)

**Comment by 1c3t3a** [comment]
Just need to fix the doctests and we're good to go! :)

**Comment by 1c3t3a** [comment]
It would be great to get this into the new version! If the three comments I posted are resolved and all tests pass I am happy to merge.

**Comment by shenjackyuanjie** [comment]
@1c3t3a please approval the test, I think it will do it

**Comment by shenjackyuanjie** [comment]
hmmm, A quick fix

**Comment by shenjackyuanjie** [comment]
19 min "quick fix"
Just try this

**Comment by 1c3t3a** [comment]
I think this won't work either, the issue is as I pointed out, that we're missing an async main function. Please take a look at my comment regarding wrapping the code in the comment.

**Comment by shenjackyuanjie** [comment]
> I think this won't work either, the issue is as I pointed out, that we're missing an async main function. Please take a look at my comment regarding wrapping the code in the comment.

oh, sorry about that, mybad

**Comment by shenjackyuanjie** [comment]
what……
fine……

**Comment by shenjackyuanjie** [comment]
image

I have tested it locally, looking good

**Comment by shenjackyuanjie** [comment]
@1c3t3a one last time today, please approve the test

**Comment by shenjackyuanjie** [comment]
finally！

today is a good day~

**Comment by 1c3t3a** [comment]
Great! Finally passed :) Could you please remove the change of base64 version please? Then we're ready to merge!

File: socketio/Cargo.toml [hunk 3]
```diff
@@ -12,7 +12,7 @@ license = "MIT"
 
 [dependencies]
 rust_engineio = { version = "0.4.3", path = "../engineio" }
-base64 = "0.21.5"
+base64 = "0.22.0"
```
Comment (1c3t3a) [inline socketio/Cargo.toml:15]: I don't really see why this is necessary?

**Comment by shenjackyuanjie** [comment]
@1c3t3a there you go~

**Review by 1c3t3a — APPROVED** [review]
LGTM, thank you very much! :)

**Pull request merged**